### PR TITLE
Replace P3A usages of `NOTREACHED_IN_MIGRATION`

### PR DESCRIPTION
**Pull request by DJAndries**
Resolves https://github.com/brave/brave-browser/issues/41898

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_news/browser/brave_news_p3a.cc [hunk 1]
```diff
@@ -165,7 +164,7 @@ void NewsMetrics::RecordTotalActionCount(ActionType action,
                                          kSidebarFilterUsageBuckets, total);
       break;
     default:
```
Comment (cdesouza-chromium) [inline components/brave_news/browser/brave_news_p3a.cc:166]: Is this a case here and in other similar places that there are no other values, and the switch is already covering all cases? if that's so, shouldn't we remove also the `default:` entry, so the compiler can alert us if a new key gets added to the switch?

Comment (DJAndries) [reply components/brave_news/browser/brave_news_p3a.cc]: yes, i think that's better. changed where applicable.

File: components/p3a/p3a_service.cc [hunk 2]
```diff
@@ -270,8 +270,7 @@ void P3AService::OnHistogramChanged(const char* histogram_name,
   const bool ok = samples->Iterator()->GetBucketIndex(&bucket);
   if (!ok) {
     LOG(ERROR) << "Only linear histograms are supported at the moment!";
-    NOTREACHED_IN_MIGRATION();
-    return;
+    NOTREACHED();
```
Comment (cdesouza-chromium) [inline components/p3a/p3a_service.cc:273]:
```cxx
NOTREACHED() << "Only linear histograms are supported at the moment!";
```

**Review by cdesouza-chromium — APPROVED** [review]
LGTM % nit.

Thanks Darnell for you time looking into this.

File: components/brave_wallet/browser/brave_wallet_p3a.cc [hunk 3]
```diff
@@ -180,8 +180,6 @@ void BraveWalletP3A::ReportJSProvider(mojom::JSProviderType provider_type,
                      : JSProviderAnswer::kNativeNotOverridden;
       }
       break;
-    default:
-      NOTREACHED_NORETURN();
```
Comment (darkdh) [inline components/brave_wallet/browser/brave_wallet_p3a.cc:184]: shouldn't this also early return by default? same as two mojom::CoinType below.

Comment (DJAndries) [reply components/brave_wallet/browser/brave_wallet_p3a.cc:184]: following the suggestion above: https://github.com/brave/brave-core/pull/26247#discussion_r1817420831

Comment (darkdh) [reply components/brave_wallet/browser/brave_wallet_p3a.cc:184]: I agree with that, what about the other two return in this file?

Comment (darkdh) [reply components/brave_wallet/browser/brave_wallet_p3a.cc:184]: oh, I see. we only choose two from that type for first one and another one is else condition

**Review by darkdh — APPROVED** [review]

File: components/p3a/p3a_service.cc [hunk 4]
```diff
@@ -269,9 +269,7 @@ void P3AService::OnHistogramChanged(const char* histogram_name,
   size_t bucket = 0u;
   const bool ok = samples->Iterator()->GetBucketIndex(&bucket);
   if (!ok) {
-    LOG(ERROR) << "Only linear histograms are supported at the moment!";
-    NOTREACHED_IN_MIGRATION();
-    return;
+    NOTREACHED() << "Only linear histograms are supported at the moment!";
```
Comment (iefremov) [inline components/p3a/p3a_service.cc:272]: i'd rather left LOG(ERROR) + return, just in case

File: components/p3a/metric_log_store.cc [hunk 5]
```diff
@@ -277,7 +274,7 @@ void MetricLogStore::DiscardStagedLog(std::string_view reason) {
 void MetricLogStore::MarkStagedLogAsSent() {}
 
 void MetricLogStore::TrimAndPersistUnsentLogs(bool overwrite_in_memory_store) {
```
Comment (iefremov) [inline components/p3a/metric_log_store.cc:276]: not for this pr, but we can refactor out obsolete inheritance and drop unused methods

File: components/p3a/histograms_braveizer.cc [hunk 6]
```diff
@@ -71,7 +71,7 @@ void HistogramsBraveizer::DoHistogramBravetization(
         answer = 1;
         break;
       default:
-        NOTREACHED_IN_MIGRATION();
+        NOTREACHED();
```
Comment (iefremov) [inline components/p3a/histograms_braveizer.cc:74]: it looks dangerous, upstream can potentially add new values to the historgram, let's not crash

File: components/misc_metrics/menu_metrics.cc [hunk 7]
```diff
@@ -37,7 +37,7 @@ const char* GetMenuGroupPrefKey(MenuGroup group) {
     case MenuGroup::kBrowserViews:
       return kBrowserViewsPrefKey;
   }
-  NOTREACHED_IN_MIGRATION();
+  return nullptr;
```
Comment (iefremov) [inline components/misc_metrics/menu_metrics.cc:40]: why drop notreached?

Comment (DJAndries) [reply components/misc_metrics/menu_metrics.cc:40]: there is no default case in the switch, so any addition to the `MenuGroup` enum will force the dev to make a subsequent addition here

File: components/brave_wallet/browser/brave_wallet_p3a.cc [hunk 8]
```diff
@@ -309,9 +304,6 @@ void BraveWalletP3A::RecordActiveWalletCount(int count,
     case mojom::CoinType::ZEC:
       histogram_name = kZecActiveAccountHistogramName;
       break;
-    default:
-      NOTREACHED_IN_MIGRATION() << coin_type;
-      return;
   }
```
Comment (iefremov) [inline components/brave_wallet/browser/brave_wallet_p3a.cc:307]: why not drop `default:` but keep NOTREACHED in the last line of the function? just like the guide recommends https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/checks.md#examples
and above

File: components/p3a/rotation_scheduler.cc [hunk 9]
```diff
@@ -104,9 +103,8 @@ const char* GetConstellationRotationTimestampPref(MetricLogType log_type) {
     case MetricLogType::kExpress:
       return kLastExpressConstellationRotationTimeStampPref;
     default:
-      NOTREACHED_IN_MIGRATION();
+      NOTREACHED();
   }
-  return nullptr;
```
Comment (iefremov) [inline components/p3a/rotation_scheduler.cc:109]: in this file, shouldn't we follow the pattern from the guide
```
// Good:
//
// This switch handles all cases, but enums can technically hold any integer
// value (even if all enum members are enumerated), so the compiler must try to
// handle other cases too. We can avoid dealing with values outside enums by
// using NOTREACHED() while also making sure we hear about it.
switch (my_enum) {
  case A: return 1;
  case B: return 5;
  case C: return 3;
}
NOTREACHED();
```

**Review by iefremov — APPROVED** [review]

File: components/p3a/rotation_scheduler.cc (hunk 9 above)
Comment (DJAndries) [reply components/p3a/rotation_scheduler.cc:109]: forgot to update, thanks

**Comment by cdesouza-chromium** [comment]
Is there anything still blocking this PR?

**Pull request merged**